File: java/src/entity/Matiere.java
package entity;

public class Matiere {

    private String nom;
    private String description;

    public Matiere(String nom, String description) {
        this.nom = nom;
        this.description = description;
    }

    // Getters
    public String getNom() { return nom; }
    public String getDescription() { return description; }

    @Override
    public String toString() {
        return nom + " (" + description + ")";
    }
}

File: php/src/entity/Matiere.php
<?php

namespace App\Entity;

class Matiere
{
    private int $id;
    private String $nom;
    private String $description;

    public function __construct() {}

    public function create(String $nom, String $description): Matiere
    {
        $mat = new Matiere();
        $mat->setNom($nom);
        $mat->setDescription($description);
        return $mat;
    }

    // Getters
    public function getId(): int
    {
        return $this->id;
    }
    public function getNom()
    {
        return $this->nom;
    }
    public function getDescription()
    {
        return $this->description;
    }

    // Setters
    public function setNom(string $nom)
    {
        $this->nom = $nom;
    }
    public function setDescription(string $description)
    {
        $this->description = $description;
    }

    //methodes complementaires
    public function toDbArray(): array
    {
        return [
            "nom" => $this->nom,
            "description" => $this->description,
        ];
    }
}
